lib/chewy/query.rb

```diff
@@ -992,7 +997,7 @@ def _response
           begin
             Chewy.client.search(_request)
           rescue Elasticsearch::Transport::Transport::Errors::NotFound => e
-            raise e if e.message !~ /IndexMissingException/
+            raise e if e.message !~ /IndexMissingException/ && e.message !~ /index_not_found_exception/
```

Comment:
If you look at [this PR](https://github.com/elastic/elasticsearch/pull/12023/files), you will see that IndexMissingException was changed to IndexNotFoundException.